source/exercises/taper.md

```diff
@@ -1,2 +1,143 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.0
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
 # 波形尖灭
 
+- 本节贡献者: {{何星辰}}、{{田冬冬}}、{{姚家园}}
+- 最近更新日期: 2025-10-22
+- 预计花费时间: 20 分钟
+
+---
+波形尖灭是指在信号首尾施加平滑衰减窗函数，使振幅在两端平滑过渡至零，从而减少截断引起的频谱泄漏。设原始信号为 $x(j)$、窗函数为 $w(j)$，则尖灭后的信号可表示为：
+
+$$
+x'(j) = w(j)*x(j),
+$$
+
+其中 $w(j)$ 在信号首尾逐渐从 0 过渡到 1，并在中间区域保持为 1，实现波形的尖灭。
+
+
+我们以前一节使用的 2022 年 9 月 22 日墨西哥 Mw 6.8 地震在 ANMO 台站的波形为例。
+
+```{code-cell} ipython3
+from obspy import UTCDateTime
+from obspy.clients.fdsn import Client
+import matplotlib.pyplot as plt
+from numpy.polynomial import polynomial as P
+
+client = Client("IRIS") 
+
+# 下载 2022 年墨西哥 Mw 6.8 级地震在 ANMO 台站的波形数据
+starttime = UTCDateTime("2022-09-22T06:18:00")
+endtime = starttime + 720  # 下载 12 分钟数据
+st = client.get_waveforms(
+    network="IU",
+    station="ANMO", 
+    location="00", 
+    channel="BHZ",
+    starttime=starttime, 
+    endtime=endtime,
+)
+st.plot();
+```
+
+进行去波形尖灭处理之前，通常需要先进行去均值、去线性趋势操作。
+
+```{code-cell} ipython3
+tr = st[0]
+
+# 去均值 + 去趋势
+tr.detrend("demean")
+tr.detrend("linear")
+
+# 创建副本
+tr_prev = tr.copy()  # 尖灭前波形
+tr_proc = tr.copy()  # 尖灭后波形
+```
+
+波形尖灭可使用 ObsPy 的 {meth}`obspy.core.trace.Trace.taper` 实现，下面是常用的三种尖灭窗函数及其参数：
+
+|    类型   |     $\omega$     | $F_0$ | $F_1$ |
+| :-----: | :--------------: | :---: | :---: |
+| HANNING |  $\frac{\pi}{N}$ |  0.50 |  0.50 |
+| HAMMING |  $\frac{\pi}{N}$ |  0.54 |  0.46 |
+|  COSINE | $\frac{\pi}{2N}$ |  1.00 |  1.00 |
+
+:::{figure} taper-functions.png
+:align: center
+:alt: "taper 衰减函数曲线"
+:width: 95%
+
+Taper 衰减曲线（引自 [SAC中文手册](https://seisman.github.io/SAC_Docs_zh/commands/taper/)，图 25）。
+:::
```

Comment:
Please remove this figure and use this one instead. 
<img width="989" height="490" alt="image" src="https://github.com/user-attachments/assets/ee713b21-dd60-4451-b973-76ae2a8bc0ef" />